pkg/deploy/deploy.go

```diff
@@ -43,21 +43,35 @@ func Deploy(workingDir string, fileReader util.FileReader, environmentsFile stri
 	}
 
 	for _, environment := range environments {
-		err := execute(environment, projects, dryRun, workingDir)
-		if err != nil {
-			deploymentErrors[environment.GetId()] = err
+		errors := execute(environment, projects, dryRun, workingDir, continueOnError)
+		if errors != nil {
+			deploymentErrors[environment.GetId()] = errors
 		}
 	}
 
 	for environment, err := range deploymentErrors {
 		if dryRun {
 			util.Log.Error("Validation of %s failed with error %s\n", environment, err)
-			util.PrettyPrintError(err)
+			for _, envError := range err {
+				util.PrettyPrintError(envError)
```

Comment:
I guess we now also have to change the `PrettyPrintError` function to have a 
fallback and print just the error. We might even consider to introduce a `PrettyPrintable`
interface and try to cast to this, instead of concrete types in there.